pkg/mutation/path/parser/parser_test.go

```diff
@@ -0,0 +1,284 @@
+				&Object{Reference: "securityContext"},
+			},
+		},
+		{
+			input: `spec.containers[name: foo].securityContext`,
+			expected: []Node{
+				&Object{Reference: "spec"},
+				&Object{Reference: "containers"},
+				&List{KeyField: "name", KeyValue: strPtr("foo")},
+				&Object{Reference: "securityContext"},
+			},
+		},
+		{
+			input: `spec.containers["my key": "foo bar"]`,
+			expected: []Node{
+				&Object{Reference: "spec"},
+				&Object{Reference: "containers"},
+				&List{KeyField: "my key", KeyValue: strPtr("foo bar")},
+			},
+		},
+		{
+			// Error: keys with whitespace must be quoted
+			input:     `spec.containers[my key: "foo bar"]`,
+			expectErr: true,
+		},
+		{
+			// Error: values with whitespace must be quoted
+			input:     `spec.containers[key: foo bar]`,
+			expectErr: true,
+		},
+		{
+			input: `spec.containers[name: ""].securityContext`,
+			expected: []Node{
+				&Object{Reference: "spec"},
+				&Object{Reference: "containers"},
+				&List{KeyField: "name", KeyValue: strPtr("")},
+				&Object{Reference: "securityContext"},
+			},
+		},
+		{
+			input: `spec.containers["": "someValue"].securityContext`,
+			expected: []Node{
+				&Object{Reference: "spec"},
+				&Object{Reference: "containers"},
+				&List{KeyField: "", KeyValue: strPtr("someValue")},
+				&Object{Reference: "securityContext"},
+			},
+		},
+		{
+			// Parsing error: either glob or field value are required in listSpec
+			input:     `spec.containers[name: ].securityContext`,
+			expectErr: true,
+		},
+		{
+			// parse error: listSpec requires keyField
+			input:     `spec.containers[].securityContext`,
+			expectErr: true,
+		},
+		{
+			input:     `spec.containers[:].securityContext`,
+			expectErr: true,
+		},
+		{
+			input:     `spec.containers[:foo].securityContext`,
+			expectErr: true,
+		},
+		{
+			input:     `spec.containers[foo].securityContext`,
+			expectErr: true,
+		},
+		{
+			input:     `spec.containers[*].securityContext`,
+			expectErr: true,
+		},
+		{
+			// parse error: we don't allow empty segments
+			input:     `foo..bar`,
+			expectErr: true,
+		},
+		{
+			// ...but we do allow zero-string-named segments
+			input: `foo."".bar`,
+			expected: []Node{
+				&Object{Reference: "foo"},
+				&Object{Reference: ""},
+				&Object{Reference: "bar"},
+			},
+		},
+		{
+			// whitespace can surround tokens
+			input: `    spec   .    containers    `,
+			expected: []Node{
+				&Object{Reference: "spec"},
+				&Object{Reference: "containers"},
+			},
+		},
+		{
+			// whitespace can surround tokens
+			input: `    spec   .    "containers"    `,
+			expected: []Node{
+				&Object{Reference: "spec"},
+				&Object{Reference: "containers"},
+			},
+		},
+		{
+			// List cannot start the path
+			input:     `[foo: bar]`,
+			expectErr: true,
+		},
+		{
+			// List cannot follow list
+			input:     `[foo: bar][bar: *]`,
+			expectErr: true,
+		},
+		{
+			// List cannot follow valid list
```

Comment:
Do we also want to check pathentries that end in a globbed list? That should also be disallowed, as then we would be setting multiple objects to the exact same value.

Reply:
Hmm - I'm not sure that be a parse error, but perhaps a validation to apply at a higher level?

Reply:
That's a fair thought, I think the path tests will allow glob-terminated locations.